daemon/logger/ring.go

```diff
@@ -0,0 +1,219 @@
+package logger
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+const (
+	defaultRingMaxSize = 1000000 // 1MB
+)
+
+// RingLogger is a ring buffer that implements the Logger interface.
+// This is used when lossy logging is OK.
+type RingLogger struct {
+	buffer *messageRing
+	ctx    context.Context
```

Comment:
If we look closer on ctx usage here, we'll see that we don't even need channel. It can be variable with mutex. At least I don't see real selects which we can cancel with context or channel.